homeworks/12/Instagram/ServletPosts.java

```diff
@@ -0,0 +1,52 @@
+package org.elsys.hwservlet;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonObject;
+import jakarta.servlet.*;
+import jakarta.servlet.http.*;
+import jakarta.servlet.annotation.*;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.Map;
+
+@WebServlet(name = "ServletPosts", value = "/posts/*")
+public class ServletPosts extends HttpServlet {
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        String path = request.getPathInfo();
+        String[] path_split = path.split("/");
+
+        HttpSession session = request.getSession();
+        if (session == null) {
+            response.setStatus(401);
+            return;
+        }
+        User user = (User) session.getAttribute("User");
+        if (!path_split[2].equals("comments")) {
+            response.setStatus(400);
+            return;

```

Comment:
If you `return` immediately after calling `setStatus` with an error code, you may want to use `sendError`.
```suggestion
            response.sendError(HttpServletResponse.SC_BAD_REQUEST);
```